Add tableOfContents styles

HaSistrunk opened the pull request:
Fixes #68 

Based on styles from rockarch.org. 

rockarch.org removes the table of contents display on small screens, where you can then access it using a button. From the html, it looks like we were looking to move away from that, or at least remove it from the component here.

Storybook includes 3 color versions and one with a title, which we don't have existing styles for, so I made some design decisions on those. I welcome feedback if others have a different vision.

I also removed the [selector-psuedo-class-focus](https://github.com/YozhikM/stylelint-a11y/blob/master/src/rules/selector-pseudo-class-focus/README.md) rule from stylelint/a11y because it wasn't letting me adjust hover styles without also applying a focus style (even though there were existing focus styles that had already been defined in _base.scss).

HaSistrunk committed:
add styles from rockarch.org with linting

HaSistrunk committed:
update styles

HaSistrunk committed:
add title styles and remove focus linting

HaSistrunk committed:
add comments

HaSistrunk committed:
don't remove ariaLabel control

HaSistrunk committed:
add color contrast to known issues

helrond commented on .stylelintrc.json:
```diff
@@ -9,7 +9,7 @@
     "a11y/media-prefers-reduced-motion": true,
     "a11y/no-outline-none": true,
-    "a11y/selector-pseudo-class-focus": true,
+    "a11y/selector-pseudo-class-focus": null,
```
I think you can just remove this line to disable. IIRC the way the a11y plugin works is that you have to explicitly enable rules.

helrond reviewed (changes requested):
See my comment below about disabling `a11y/selector-pseudo-class-focus`. 

Because of the `sticky` class, the different colors of the TOC kind of collapse on top of each other when scrolling. This doesn't happen with the TOC with a title, so I'm wondering if putting a header between each of them, or adding a wrapping div would help?

The TOC with a title was intended for use on the docs site - I think what you've done is totally fine; we can adjust if necessary when implementing.

helrond commented:
> rockarch.org removes the table of contents display on small screens, where you can then access it using a button. From the html, it looks like we were looking to move away from that, or at least remove it from the component here.

We still need a strategy for mobile though, right? I think part of the complexity is that the actual HTML structure for the table of contents is created by JavaScript by looking at the page headers.

bonniegee commented:
I think the line spacing on the title when it wraps (such as on small mobile) may be a little tight: <img width="312" alt="Screen Shot 2021-07-21 at 10 16 31" src="https://user-images.githubusercontent.com/8737364/126503892-c9a10f62-c33d-4c0d-ab66-4cfb1db14357.png">

HaSistrunk committed:
remove stylelint rule

HaSistrunk committed:
only apply line-height change to list

HaSistrunk commented:
1. In terms of a solution for mobile, I can only really think to build back in the button from rockarch.org since that's the design we have. Otherwise we could just have it move from a multi-column layout on large screens to having the toc take the full width of the screen on small screens like it does on the docs site.
2. I was trying to use [Storybook decorators](https://storybook.js.org/docs/react/writing-stories/decorators) in the `.stories.js` file to wrap the color components in a div (which does keep them from overlapping from the sticky styles), but wasn't able to make it work. @helrond , did you ever try to use decorators with components? I can add a wrapping `<div>` in the handlebars file, but was trying to avoid messing with the component html.

helrond committed:
create individual exports for each color

HaSistrunk committed:
Merge pull request #88 from RockefellerArchiveCenter/issue-68-wrap

Create individual exports for each tableOfContents color

helrond commented:
I think my instinct is to keep this component simple. The reality is that the button/dropdown is a completely different component, and (I think) is implemented so that it's shown in place of the table of contents on a smaller screen.

HaSistrunk commented:
> I think my instinct is to keep this component simple. The reality is that the button/dropdown is a completely different component, and (I think) is implemented so that it's shown in place of the table of contents on a smaller screen.

Seems like a good path forward. We can iterate later if necessary. I think I still need to do a little cleanup here, though.

HaSistrunk committed:
move width property

HaSistrunk committed:
Merge branch 'development' into issue-68

HaSistrunk committed:
fix imports

helrond commented on stories/tableOfContents/tableOfContents.handlebars:
```diff
@@ -1,10 +1,10 @@
-  <a class="toc__title" href="/digital-preservation-policy/">{{title}}</a>
-  {{/if}}
   <div class="toc__list">
+    {{#if title}}
```
What's the logic in dropping this inside the `toc__list` class, and is there any point in retaining that wrapping div?

HaSistrunk replied:
I dropped it inside to more easily adjust styles for a group of a title and associated list items together. But I've dropped that div and the `toc__list` entirely now to keep it simple. If we need to add divs later for layout, we can do that when we implement.

helrond commented on stories/tableOfContents/listItems.json:
```diff
@@ -1,13 +1,13 @@
-  {"text": "Roles and Responsibilities", "active": false},
-  {"text": "Appendix A: Glossary", "active": false},
-  {"text": "Appendix B: Sources Consulted", "active": false}
+  {"text": "Purpose"},
+  {"text": "Objectives"},
+  {"text": "Mandate"},
+  {"text": "Scope"},
+  {"text": "Challenges"},
+  {"text": "Principles"},
+  {"text": "Categories of Commitment"},
+  {"text": "Frequency of Policy Review"},
+  {"text": "Roles and Responsibilities"},
+  {"text": "Appendix A: Glossary"},
+  {"text": "Appendix B: Sources Consulted"}
```
Can you explain more about removing the `active` class and then adding a style for `a:active` below?

HaSistrunk replied:
I was thinking that the `:active` pseudo-class should cover it without it needing to be its own class. Did I overlook the reason for that approach?

helrond replied:
Hmm I am wondering if we added an `.active` class so we could style the ToC item that is visible on the page. There's some styling like this on the docs site...

HaSistrunk committed:
include title

HaSistrunk committed:
include active class and enable multiple titles

HaSistrunk committed:
Merge branch 'development' into issue-68

HaSistrunk commented:
Ok, I've simplified this a bit, added the `active` class, and tweaked things to better enable multiple titles to be used (like in the docs site). I also updated the Storybook documentation accordingly.

helrond reviewed (approved):
Looks good!

HaSistrunk committed:
actually change padding amount with screen size
